File: update_data_model/Visibility_Docs_AI_VM-feature-folder-tree-view/Visibility_Docs_AI_VM-feature-folder-tree-view/backend/app/services/email_service.py
import smtplib
import logging
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from ..config import settings

logger = logging.getLogger("visibility-docs")


def send_email(to_email: str, subject: str, html_body: str) -> bool:
    host = settings.EMAIL_HOST
    port = settings.EMAIL_PORT
    username = settings.EMAIL_USERNAME
    password = settings.EMAIL_PASSWORD
    from_addr = settings.EMAIL_FROM

    if not username or not password:
        logger.warning("Email not configured — skipping send")
        return False

    msg = MIMEMultipart("alternative")
    msg["Subject"] = subject
    msg["From"] = from_addr or username
    msg["To"] = to_email
    msg.attach(MIMEText(html_body, "html"))

    try:
        with smtplib.SMTP(host, port) as server:
            server.starttls()
            server.login(username, password)
            server.send_message(msg)
        logger.info(f"Email sent to {to_email}: {subject}")
        return True
    except Exception as e:
        logger.error(f"Failed to send email to {to_email}: {e}")
        return False

File: api-gateway/src/services/emailService.ts
import nodemailer from 'nodemailer';
import logger from '../utils/logger';

export function isEmailConfigured(): boolean {
    return Boolean(
        String(process.env.EMAIL_USERNAME || '').trim() &&
            String(process.env.EMAIL_PASSWORD || '').trim()
    );
}

export async function sendHtmlEmail(params: {
    to: string | string[];
    subject: string;
    html: string;
}): Promise<void> {
    const host = process.env.EMAIL_HOST || 'smtp.gmail.com';
    const port = Number(process.env.EMAIL_PORT || 587);
    const username = String(process.env.EMAIL_USERNAME || '').trim();
    const password = String(process.env.EMAIL_PASSWORD || '').trim();
    const from = String(process.env.EMAIL_FROM || username || '').trim();

    if (!username || !password) {
        throw Object.assign(
            new Error(
                'Email is not configured. Set EMAIL_USERNAME and EMAIL_PASSWORD (and optionally EMAIL_HOST, EMAIL_PORT, EMAIL_FROM) on the api-gateway.'
            ),
            { statusCode: 400 }
        );
    }

    const recipients = (Array.isArray(params.to) ? params.to : [params.to])
        .map((e) => String(e || '').trim())
        .filter(Boolean);
    if (!recipients.length) {
        throw Object.assign(new Error('No recipients specified'), { statusCode: 400 });
    }

    const transporter = nodemailer.createTransport({
        host,
        port,
        secure: port === 465,
        auth: { user: username, pass: password },
    });

    await transporter.sendMail({
        from: from || username,
        to: recipients.join(', '),
        subject: params.subject,
        html: params.html,
    });

    logger.info(`[email] Sent "${params.subject}" to ${recipients.join(', ')}`);
}
